### PR TITLE
CI: improve project tests logging

### PR DESCRIPTION
**Pull request by eli-schwartz**
/cc @xclaesse

### CODE REVIEW

**Comment by eli-schwartz** [comment]
Example output: https://github.com/eli-schwartz/meson/actions/runs/6029933827/job/16360634546

File: run_project_tests.py [hunk 1]
```diff
@@ -1622,17 +1623,21 @@ def clear_transitive_files() -> None:
         (passing_tests, failing_tests, skipped_tests) = res
     except StopException:
         pass
-    print()
-    print('Total passed tests: ', green(str(passing_tests)))
-    print('Total failed tests: ', red(str(failing_tests)))
-    print('Total skipped tests:', yellow(str(skipped_tests)))
     if failing_tests > 0:
         print('\nMesonlogs of failing tests\n')
```
Comment (xclaesse) [inline run_project_tests.py:1627]: I see you kep this line, but I don't see it in the logs you pasted. Shouldn't this be deleted?

**Comment by jpakkane** [comment]
Looking nice. Would it be possible to make the Python unittest thingy output the brief dot format? Having a ton of lines like these is not very informative:

```
 unittests/machinefiletests.py::NativeFileTests::test_builtin_options_subprojects_overrides_buildfiles 
[gw1] [ 90%] PASSED unittests/machinefiletests.py::NativeFileTests::test_builtin_options_subprojects_overrides_buildfiles 
```

File: run_project_tests.py (hunk 1 above)
Comment (eli-schwartz) [reply run_project_tests.py:1627]: I do see it though:
```
 [SKIPPED]   wayland: 2 core only

Total configuration time: 1877.07s
Total build time:         854.80s
Total test time:          76.47s

Mesonlogs of failing tests


=========================================== common: 2 cpp ==========================================
 

(inprocess) $ setup --prefix /usr --libdir lib 'test cases/common/2 cpp' '/cygdrive/d/a/meson/meson/b 1f81195594' --backend=ninja
```

**Comment by eli-schwartz** [comment]
> While at it, something I've always wanted to change is tests that fails because there are unexepected installed files, the list of files is not printed in the summary of failed tests, you have to scroll all the way up to when it ran the test.

Implemented in 71f5a1241b82393bf07551349ab8a0b3cb3aa675

Example log: https://github.com/eli-schwartz/meson/actions/runs/6042585162/job/16397982654#step:8:2447

File: run_project_tests.py (hunk 1 above)
Comment (xclaesse) [reply run_project_tests.py:1627]: Oh sorry, that's indeed before the failed tests logs.

**Comment by xclaesse** [comment]
LGTM, nice work!

**Comment by jpakkane** [comment]
FWICT Pytest still prints the long format for succeeding tests:

```
 unittests/platformagnostictests.py::PlatformAgnosticTests::test_setup_loaded_modules 
[gw0] [ 86%] SKIPPED unittests/platformagnostictests.py::PlatformAgnosticTests::test_setup_loaded_modules 
```

**Comment by jpakkane** [comment]
It's not mandatory, so ok to merge, but I feel it would improve readability.

**Comment by eli-schwartz** [comment]
Changing that would have been an additional commit, but I haven't yet looked at what, if anything, it is doing there.

**Pull request merged**